File: Dynamic Programming/Climbing_stairs.cpp
//Problem link: https://leetcode.com/problems/climbing-stairs/
//TC: O(N) SC: O(N)
class Solution {
public:
    int climbStairs(int n) {
        if(n==1) return 1;
        int dp[n+1];
        dp[1]=1;
        dp[2]=2;
        for(int i=3;i<=n;i++){
            dp[i]=dp[i-1]+dp[i-2];
        }
        return dp[n];
    }
};

File: Dynamic Programming/Fibonacci DP/Climbing stairs.java
//Prob link: https://leetcode.com/problems/climbing-stairs
//TC: O(N) SC: O(1)
class Solution {
    public int climbStairs(int n) {
        
        if(n==1) return 1;
        int prev2=1;
        int prev1=2;
        int cur;
        for(int i=3;i<=n;i++){
            cur=prev1+prev2;
            prev2=prev1;
            prev1=cur;
        }
        return prev1;
        
    }
}
